### Peer Review Activity 3.2.9

### Below are my codes with insturctions before each code

In [3]:
# make sure the data file is uploaded properly
# import pandas
import pandas as pd

# read CSV file from the current working directory
products = pd.read_csv('products.csv')

products.shape

(3945, 2)

In [4]:
# create a user-defined function to view products contain the word 'fire'
def contains_fire(x):
    """Does the product contain the word 'fire'?"""
    y = x.lower()
    return 'fire' in y

# now apply the function created with the apply() function to answer the question
cf = products['Description'].apply(contains_fire)

# view and filter
print(cf)
products[cf]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                          Description
1318     22428             ENAMEL FIRE BUCKET CREAM
3033    84569B       PACK 3 FIRE ENGINE/CAR PATCHES
3626    90039A     FIRE POLISHED GLASS BRACELET RED
3627    90039B  FIRE POLISHED GLASS BRACELET MONTAN
3628    90039C   FIRE POLISHED GLASS BRACELET BLACK
3629    90039D   FIRE POLISHED GLASS BRACELET GREEN
3656    90060B       FIRE POLISHED GLASS NECKL GOLD
3657    90060D     FIRE POLISHED GLASS NECKL BRONZE
3658    90060F      FIRE POLISHED GLASS NECKL GREEN

#### by counting above, the answer for the questions is 9 products contain the word fire

In [6]:
# repeat the same process with creating a user-defined function to view products contain the word 'candle'
def contains_candle(x):
    """Does the product contain the word 'candle'?"""
    y = x.lower()
    return 'candle' in y

# now apply the function created with the apply() function to answer the question
cc = products['Description'].apply(contains_candle)

# view and filter
print(cc)
products[cc]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                         Description
83      17012A  ORIGAMI VANILLA INCENSE/CANDLE SET
84      17012B  ORIGAMI JASMINE INCENSE/CANDLE SET
86      17012D     ORIGAMI ROSE INCENSE/CANDLE SET
87      17012E    ORIGAMI OPIUM INCENSE/CANDLE SET
107     17136A      BLUE STONES ON WIRE FOR CANDLE
...        ...                                 ...
3530    85230A         OPIUM SCENTED VOTIVE CANDLE
3531    85230B      CINNAMON SCENTED VOTIVE CANDLE
3532    85230E      STRAWBRY SCENTED VOTIVE CANDLE
3533    85230F               JASMINE VOTIVE CANDLE
3534    85230G                ORANGE VOTIVE CANDLE

[178 rows x 2 columns]

####  By checking the total rows above, the answer for the questions is 178 products contain the word candle

In [7]:
# repeat the same process with creating a user-defined function to view products contain the word 'matches'
def contains_matches(x):
    """Does the product contain the word 'matches'?"""
    y = x.lower()
    return 'matches' in y

# now apply the function created with the apply() function to answer the question
cm = products['Description'].apply(contains_matches)

# view and filter
print(cm)
products[cm]

0       False
1       False
2       False
3       False
4       False
        ...  
3940    False
3941    False
3942    False
3943    False
3944    False
Name: Description, Length: 3945, dtype: bool


StockCode                      Description
669     21584     RETROSPOT SMALL TUBE MATCHES
670     21586  KINGS CHOICE GIANT TUBE MATCHES
671     21587     COSY HOUR GIANT TUBE MATCHES
672     21588     RETROSPOT GIANT TUBE MATCHES
673     21591      COSY HOUR CIGAR BOX MATCHES
674     21592      RETROSPOT CIGAR BOX MATCHES

#### by counting above, the answer for the questions is 6 products contain the word matches